2_base_and_sensor.ipynb

```diff
@@ -436,135 +812,259 @@
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "# Control the robot using sensor values"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "センサ値を使ってロボットを制御してみましょう。"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Let's control the robot using the sensor value."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "壁の1メートル手前で止まる（壁の1メートル手前まで前進する）プログラムを書いてみましょう。\n",
     "\n",
     "while文を使うと以下のように書けると思います。"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Let's write a program that stops 1 meter before the wall (go straight until the robot reaches 1meter before the wall).\n",
+    "\n",
+    "You can use `while` statement to write your code:"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "outputs": [],
    "source": [
     "while True:\n",
     "    scan_data = laser.get_data()\n",
-    "    if scan_data.ranges[360] < 1.0:  # 1メートル以内になったらwhileループを抜ける\n",
+    "    if scan_data.ranges[360] < 1.0:  # 1メートル以内になったらwhileループを抜ける Exit the while loop if within 1 meter\n",
     "        break\n",
     "        \n",
     "    move_base(0.5, 0, 0)"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "<div class=\"alert alert-block alert-info\">\n",
     "    <b>課題</b>\n",
     "    <p>初期位置から人の前まで移動するプログラムを書いてみましょう。今回はレーザの値を用いましょう。</p>\n",
     "</div>"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "<div class=\"alert alert-block alert-info\">\n",
+    "    <b>Task</b>\n",
+    "    <p>Let's write a code to move the robot from the initial position to the front of the human.</p>\n",
+    "    <p>This time, use the laser scanner value to ensure the movement.</p>\n",
+    "</div>"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "outputs": [],
    "source": [
-    "# 自分で考えてみましょう。この下に入力できます。\n"
+    "# 自分で考えてみましょう。この下に入力できます。 Write your code here:\n"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "<div class=\"alert alert-block alert-info\">\n",
     "    <b>アドバンスド課題</b>\n",
     "    <p>余裕がある人は「壁にぶつからないで部屋を動き回る」プログラムを書いてみましょう。</p>\n",
     "</div>"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "<div class=\"alert alert-block alert-info\">\n",
+    "    <b>Advanced task</b>\n",
+    "    <p>If possible, write a code to \"move around the room without hitting the wall\".</p>\n",
```

Comment:
"    <p>If possible, write code to \"move around the room without hitting the wall\".</p>\n",